### PR TITLE
FISH-5746 #5419 Fix the getOptionalValues NoSuchElementException

### PR DESCRIPTION
**Pull request by arrenping**
## Description
Fix the method getOptionalValues that it throws NoSuchElementException if the result is empty.

### CODE REVIEW

**Comment by rdebusscher** [comment]
Hi @arrenping 

Thank you for the PR! We're happy to receive your contribution, however, before we can proceed you will need to sign the CLA and send it to us.

[Payara CLA Form](https://github.com/payara/Payara/blob/master/PayaraCLA.pdf)

Regards
Rudy

**Comment by arrenping** [comment]
Hi @rdebusscher 

The CLA has sent to cla@payara.org
Please have a check.

Thanks,
Arren

**Comment by smillidge** [comment]
I can confirm the CLA has been received

**Comment by Pandrex247** [comment]
Jenkins test please

**Review by Pandrex247 — CHANGES_REQUESTED** [review]

**Review by Pandrex247 — APPROVED** [review]

**Comment by Pandrex247** [comment]
Jenkins test please

**Review by breakponchito — APPROVED** [review]
LGTM

**Comment by breakponchito** [comment]
Jenkins test

File: nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/spi/PayaraConfig.java [hunk 1]
```diff
@@ -169,7 +169,7 @@ public ConfigValue getConfigValue(String propertyName) {
     @Override
     public <T> Optional<List<T>> getOptionalValues(String propertyName, Class<T> propertyType) {
         Optional<List<T>> valuesList =  Config.super.getOptionalValues(propertyName, propertyType);
-        if (valuesList.get().isEmpty()) {
+        if (valuesList.isPresent() && valuesList.get().isEmpty()) {
```
Comment (svendiedrichsen) [inline nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/spi/PayaraConfig.java:172]: How about using Optional.filter?

Comment (arrenping) [reply nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/spi/PayaraConfig.java]: Thanks for kindly reminding that method!
I've updated the implementation.

File: nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/spi/PayaraConfig.java [hunk 2]
```diff
@@ -169,7 +169,7 @@ public ConfigValue getConfigValue(String propertyName) {
     @Override
     public <T> Optional<List<T>> getOptionalValues(String propertyName, Class<T> propertyType) {
         Optional<List<T>> valuesList =  Config.super.getOptionalValues(propertyName, propertyType);
-        if (valuesList.get().isEmpty()) {
+        if (valueList.isPresent() && valuesList.get().isEmpty()) {
```
[the remark this answers is not in the document]
Comment (arrenping) [reply nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/spi/PayaraConfig.java]: Thank you!
(My mistake...

**Comment by arrenping** [comment]
Jenkins test

**Comment by arrenping** [comment]
Jenkins test please

**Pull request merged**